src/main/scala/com/comcast/xfinity/sirius/uberstore/segmented/SegmentedUberStore.scala

```diff
@@ -97,7 +97,10 @@ object SegmentedUberStore {
  *
  * @param base directory SegmentedUberStore is based in
  */
-class SegmentedUberStore private[segmented] (base: JFile, eventsPerSegment: Long, segmentedCompactor: SegmentedCompactor) extends SiriusLog {
+class SegmentedUberStore private[segmented] (base: JFile,
+                                             eventsPerSegment: Long,
+                                             segmentedCompactor: SegmentedCompactor,
+                                             siriusConfig: SiriusConfiguration) extends SiriusLog {
```

Comment:
I'd suggest we use the same pattern of making the `buildSegment` in the `apply` function, so we're not passing the configuration into the class itself. I much prefer passing the exact parameters required by the class, and letting the apply method be the thing that extracts whatever's necessary from the config in order to build those params.